### PR TITLE
RSDK-6885 - Add fields to slam.Properties

### PR DESCRIPTION
**Pull request by kkufieta**
Ticket: https://viam.atlassian.net/browse/RSDK-6885

API PR: https://github.com/viamrobotics/api/pull/458

Done:
* Add `SensorType`
* Add to `Properties`:
  * Add `InternalStateFileType`
  * Add `SensorInfo`
* Update tests

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -397,3 +397,5 @@ require (
 	github.com/ziutek/mymysql v1.5.4 // indirect
 	golang.org/x/exp v0.0.0-20230725012225-302865e7556b
 )
+
+replace go.viam.com/api => github.com/kkufieta/api v0.0.0-20240313154703-4641e8e69fbd
```
Comment (kkufieta) [inline go.mod:401]: WIll update once API PR is merged

**Review by randhid — APPROVED** [review]

File: services/slam/slam.go [hunk 2]
```diff
@@ -167,3 +185,27 @@ func protobufToMappingMode(mappingMode pb.MappingMode) (MappingMode, error) {
...
+	case pb.SensorType_SENSOR_TYPE_CAMERA:
+		return SensorTypeCamera, nil
+	case pb.SensorType_SENSOR_TYPE_MOVEMENT_SENSOR:
+		return SensorTypeMovementSensor, nil
+	case pb.SensorType_SENSOR_TYPE_UNSPECIFIED:
+		fallthrough
+	default:
+		return 0, errors.New("sensor type unspecified")
+	}
+}
```
Comment (JohnN193) [inline services/slam/slam.go:211]: [q] should these live in the same spot as the grpchelper functions?

File: services/slam/slam.go [hunk 3]
```diff
@@ -167,3 +185,27 @@ func protobufToMappingMode(mappingMode pb.MappingMode) (MappingMode, error) {
 		return 0, errors.New("mapping mode unspecified")
 	}
 }
+
+func sensorTypeToProtobuf(sensorType SensorType) pb.SensorType {
```
Comment (JohnN193) [inline services/slam/slam.go:189]: [nit] add a description for the functions

**Review by JohnN193 — APPROVED** [review]
LGTM, just a nit

File: services/slam/slam.go (hunk 2 above)
Comment (JohnN193) [reply services/slam/slam.go]: then maybe the grpchelper functions should just live in slam.go, or should be a part of the slam package? seems weird to have two different packages define helper functions

Comment (kkufieta) [reply services/slam/slam.go]: I'll move them out to the slam package!

Comment (kkufieta) [reply services/slam/slam.go]: Done

Comment (randhid) [reply services/slam/slam.go]: we usually have all the grpc helper functions int he resource package itself. I didn't know that was not the case in slam, sorry!

**Pull request merged**